### ISSUE TITLE
test flake: `TestServiceGetJobStream_bufferedData` is considered a flaky test in circle-ci

### ISSUE THREAD

**Issue by briancain** [hashicorp/waypoint#3241]
Since this isn't failing all the time, we should keep an eye out for this failure. If someone familiar with the codebase or the tests wants to look over this test and see why this test fails that would be great! Otherwise we'll keep a lookout to see if this issue persists.

**Describe the bug**

Our test occasionally fails on Circle-CI and is considered flaky.

```
Failed
=== RUN   TestServiceGetJobStream_bufferedData
2022-04-19T21:39:06.063Z [INFO]  poll_queuer.project: starting
2022-04-19T21:39:06.063Z [INFO]  poll_queuer.application_statusreport: starting
2022-04-19T21:39:06.063Z [INFO]  prune: starting
2022-04-19T21:39:06.067Z [INFO]  waiting for job assignment: runner_id=01G11W7J6JRN730E75JM38WH8A
    service_job_test.go:918: received event, but not the type we wanted: *gen.GetJobStreamResponse_State_
    service_job_test.go:918: received event, but not the type we wanted: *gen.GetJobStreamResponse_Job
    service_job_test.go:918: received event, but not the type we wanted: *gen.GetJobStreamResponse_State_
    service_job_test.go:918: received event, but not the type we wanted: *gen.GetJobStreamResponse_Job
    service_job_test.go:694: 
        	Error Trace:	service_job_test.go:694
        	Error:      	Should be true
        	Test:       	TestServiceGetJobStream_bufferedData
2022-04-19T21:39:06.068Z [WARN]  unknown error from recvmsg: runner_id=01G11W7J6JRN730E75JM38WH8A err="rpc error: code = Canceled desc = context canceled"
2022-04-19T21:39:06.068Z [ERROR] failed to mark runner as offline. This should not happen.: runner_id=01G11W7J6JRN730E75JM38WH8A err="database not open"
--- FAIL: TestServiceGetJobStream_bufferedData (0.01s)
```

**Steps to Reproduce**

https://app.circleci.com/insights/github/hashicorp/waypoint/workflows/go-tests/tests

Occasionally, the test fails in circle. There's no specific case at the moment I can think of to make it fail!

**Expected behavior**

Less flakes, or none at all if possible.

**Comment by evanphx** [comment]
@briancain thinks that he fixed this this morning (4/20) #3245.

**Comment by briancain** [comment]
I haven't seen this test fail in a long time, 1 year later.
